### PR TITLE
fix(edgeless): add a paragraph block when clicking on empty space on the note block

### PR DESCRIPTION
**Pull request by golok727**
close https://github.com/toeverything/blocksuite/issues/6631

### CODE REVIEW

File: packages/blocks/src/root-block/edgeless/components/block-portal/note/edgeless-note.ts [hunk 1]
```diff
@@ -188,6 +189,49 @@ export class EdgelessBlockPortalNote extends EdgelessPortalBase<NoteBlockModel>
...
     handleNativeRangeAtPoint(x, y);
+
+    // add an empty paragraph
+    const readonly = this.surface.doc.readonly;
+    const last = this.model.children.at(-1);
+    if (readonly) return;
+    if (
+      !last ||
+      !last.text ||
+      matchFlavours(last, [
```
Comment (donteatfriedrice) [inline packages/blocks/src/root-block/edgeless/components/block-portal/note/edgeless-note.ts:200]: Can be changed to check whether it is a paragraph or a list

```
!matchFlavours(parent, ['affine:paragraph', 'affine:list'])
```

File: packages/blocks/src/root-block/edgeless/components/block-portal/note/edgeless-note.ts [hunk 2]
```diff
@@ -188,6 +189,49 @@ export class EdgelessBlockPortalNote extends EdgelessPortalBase<NoteBlockModel>
     const x = clamp(e.x, rect.left + offsetX, rect.right - offsetX);
     const y = clamp(e.y, rect.top + offsetY, rect.bottom - offsetY);
     handleNativeRangeAtPoint(x, y);
+
+    // add an empty paragraph
```
Comment (donteatfriedrice) [inline packages/blocks/src/root-block/edgeless/components/block-portal/note/edgeless-note.ts:193]: In this case, when the last block is a non-text block, a paragraph block will always be added at the bottom. The logic here should be to focus on the nearest text block first when clicking. If the nearest block is not text block, and the last block is not a text block, then add an empty block

**Review by donteatfriedrice — CHANGES_REQUESTED** [review]
In this case, when the last block is a non-text block, a paragraph block will always be added at the bottom. The logic here should be to focus on the nearest text block first when clicking. If the nearest block is not text block, and the last block is not a text block, then add an empty block

**Comment by golok727** [comment]
>when the last block is a non-text block, a paragraph block will always be added at the bottom. 

When I tested this, I found that a paragraph block wasn't added at the bottom as described when adding a non-text. We need to click on the empty area in the doc to add a new text block. 

https://github.com/toeverything/blocksuite/assets/123532141/747f81bd-e7b9-4ad9-a709-cc9d9aebae2d

**Comment by donteatfriedrice** [comment]
> >when the last block is a non-text block, a paragraph block will always be added at the bottom. 
> 
> 
> 
> When I tested this, I found that a paragraph block wasn't added at the bottom as described when adding a non-text. We need to click on the empty area in the doc to add a new text block. 
> 
> 
> 
> https://github.com/toeverything/blocksuite/assets/123532141/747f81bd-e7b9-4ad9-a709-cc9d9aebae2d
> 
> 
> 
> 

You can add some paragraph blocks before the table, and click at the top or left or right near the paragraph blocks, in this case, should focus on the paragraph first like in doc mode. In the video you uploaded, there are no text blocks before table,

**Comment by donteatfriedrice** [comment]
> @donteatfriedrice How is this behavior? I think we should consider adding a + button like in notion beside the drag handle
> 
>  Thing2.mp4

Looks great! Thanks! Could you add tests for it ?

**Review by donteatfriedrice — APPROVED** [review]

**Comment by golok727** [comment]
> > @donteatfriedrice How is this behavior? I think we should consider adding a + button like in notion beside the drag handle
> > Thing2.mp4
> 
> Looks great! Thanks! Could you add tests for it ?

Sure!

**Pull request merged**